contracts/button/errors.rs

```diff
@@ -0,0 +1,87 @@
+use ink_env::Error as InkEnvError;
+use ink_prelude::{format, string::String};
+use openbrush::contracts::psp22::PSP22Error;
+
+/// GameError types
+#[derive(Debug, PartialEq, Eq, scale::Encode, scale::Decode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum GameError {
+    /// Returned if reset is called before the deadline
+    BeforeDeadline,
+    /// Returned if button is pressed after the deadline
+    AfterDeadline,
+    /// Returned if a call is made from an account with missing access control privileges
+    MissingRole(String),
+    /// Returned if a call to another contract has failed
+    ContractCall(String),
+}
+
+impl From<PSP22Error> for GameError {
+    fn from(e: PSP22Error) -> Self {
+        match e {
+            PSP22Error::Custom(message) => GameError::ContractCall(message),
+            PSP22Error::InsufficientBalance => {
+                GameError::ContractCall(String::from("PSP22::InsufficientBalance"))
+            }
+            PSP22Error::InsufficientAllowance => {
+                GameError::ContractCall(String::from("PSP22::InsufficientAllowance"))
+            }
+            PSP22Error::ZeroRecipientAddress => {
+                GameError::ContractCall(String::from("PSP22::ZeroRecipientAddress"))
+            }
+            PSP22Error::ZeroSenderAddress => {
+                GameError::ContractCall(String::from("PSP22::ZeroSenderAddress"))
+            }
+            PSP22Error::SafeTransferCheckFailed(message) => {
+                GameError::ContractCall(format!("PSP22::SafeTransferCheckFailed({})", message))
+            }
+        }
+    }
+}
+
+impl From<InkEnvError> for GameError {
```

Comment:
all that can also likely be simplified to just mere:
```
    impl From<InkEnvError> for GameError {
        fn from(why: InkEnvError) -> Self {
            GameError::InkEnvError(format!("{:?}", why))
        }
    }
```

Reply:
done